### ISSUE TITLE
Static libgc.a linking is broken

### ISSUE THREAD

**Issue by djwatson** [wingo/whippet#2]
```
make
gcc -Wall -flto -fno-strict-aliasing -fvisibility=hidden -Wno-unused -O2 -g -DNDEBUG -Iapi -MMD -MP -MF .deps/mt-gcbench.bdw.gc.d -o obj/mt-gcbench.bdw.gc.o -DGC_CONSERVATIVE_ROOTS=1 -DGC_CONSERVATIVE_TRACE=1 `pkg-config --cflags bdw-gc` -include benchmarks/mt-gcbench-embedder.h -c src/bdw.c
gcc -Wall -flto -fno-strict-aliasing -fvisibility=hidden -Wno-unused -O2 -g -DNDEBUG -Iapi -MMD -MP -MF .deps/mt-gcbench.bdw.d -o obj/mt-gcbench.bdw.o -DGC_CONSERVATIVE_ROOTS=1 -DGC_CONSERVATIVE_TRACE=1 -include api/bdw-attrs.h -c benchmarks/mt-gcbench.c
gcc -Wall -flto -fno-strict-aliasing -fvisibility=hidden -Wno-unused -O2 -g -DNDEBUG -Iapi -MMD -MP -MF .deps/mt-gcbench.gc-ephemeron.d -o obj/mt-gcbench.gc-ephemeron.o -include benchmarks/mt-gcbench-embedder.h -c src/gc-ephemeron.c
gcc -lpthread -flto=auto -o bin/mt-gcbench.bdw `pkg-config --libs bdw-gc`  obj/mt-gcbench.bdw.gc.o obj/mt-gcbench.bdw.o obj/gc-stack.o obj/gc-options.o obj/gc-platform.o obj/mt-gcbench.gc-ephemeron.o
/usr/bin/ld: /tmp/ccOUkpVe.ltrans0.ltrans.o: in function `gc_allocate_slow':
/home/davejwatson/projects/whippet/src/bdw.c:86: undefined reference to `GC_generic_malloc_many'
/usr/bin/ld: /tmp/ccOUkpVe.ltrans0.ltrans.o:/home/davejwatson/projects/whippet/src/bdw.c:89: undefined reference to `GC_get_heap_size'
/usr/bin/ld: /tmp/ccOUkpVe.ltrans0.ltrans.o: in function `mark_heap':
/home/davejwatson/projects/whippet/./benchmarks/simple-gc-embedder.h:68: undefined reference to `GC_least_plausible_heap_addr'
/usr/bin/ld: /tmp/ccOUkpVe.ltrans0.ltrans.o: in function `mark_heap':
```

I think -lgc is picking up libgc.a for some reason instead of libgc.so, and since command line order matters for .a/.o files, we get undefined references.  Adding -lgc last fixes it:

```
$ gcc -lpthread -flto=auto -o bin/quads.bdw  -lgc obj/quads.bdw.gc.o obj/quads.bdw.o obj/gc-stack.o obj/gc-options.o obj/gc-platform.o obj/quads.gc-ephemeron.o `pkg-config --libs bdw-gc`
$ echo $?
0
```

**Comment by djwatson** [comment]
Aaand now I see someone already submitted a pull to fix in #1

**Comment by wingo** [comment]
Fixed.  Thanks for the reminder.
